### PR TITLE
AVX512VL eltwise: Unary operators enabled

### PR DESCRIPTION
**Pull request by deeptiag1**

### CODE REVIEW

**Comment by alheinecke** [comment]
@egeor : let's see how that breaks gemm_ext as we also want to fuse when using AVX512_VL

**Comment by deeptiag1** [comment]
@alheinecke and @egeor masking is failing for avx512vl, I am testing it now.

**Comment by deeptiag1** [comment]
@alheinecke all testing is passing now. Please review

File: src/generator_gemm_amx_microkernel.c [hunk 1]
```diff
@@ -231,14 +231,15 @@ void libxsmm_generator_gemm_amx_paired_tilestore( libxsmm_generated_code*
     }
 
     if (i_micro_kernel_config->fused_sigmoid == 1) {
+      char i_vname = (io_generated_code->arch <=LIBXSMM_X86_AVX512_VL256_CPX) ? 'y' : 'z';
```
Comment (alheinecke) [inline src/generator_gemm_amx_microkernel.c:234]: This condition can be true, just set to 'z'

File: src/generator_gemm_amx_microkernel.c [hunk 2]
```diff
@@ -538,8 +539,7 @@ void libxsmm_generator_gemm_amx_single_tilestore( libxsmm_generated_code*
   unsigned int reserved_zmms       = i_micro_kernel_config->reserved_zmms;
   unsigned int fused_eltwise       = ((i_micro_kernel_config->fused_relu == 1) || (i_micro_kernel_config->fused_relu_nobitmask == 1) || (i_micro_kernel_config->fused_sigmoid == 1)) ? 1 : 0;
 
-  if (LIBXSMM_DATATYPE_F32 == LIBXSMM_GETENUM_OUT( i_xgemm_desc->datatype ) || LIBXSMM_DATATYPE_I32 == LIBXSMM_GETENUM_OUT( i_xgemm_desc->datatype )) {
-    libxsmm_x86_instruction_tile_move( io_generated_code,
+  if (LIBXSMM_DATATYPE_F32 == LIBXSMM_GETENUM_OUT( i_xgemm_desc->datatype ) || LIBXSMM_DATATYPE_I32 == LIBXSMM_GETENUM_OUT( i_xgemm_desc->datatype )) {    libxsmm_x86_instruction_tile_move( io_generated_code,
```
Comment (alheinecke) [inline src/generator_gemm_amx_microkernel.c:542]: not sure what this is

File: src/generator_gemm_amx_microkernel_emu.c [hunk 3]
```diff
@@ -186,14 +186,15 @@ void libxsmm_generator_gemm_amx_paired_tilestore_emu( libxsmm_generated_code*
     }
 
     if (i_micro_kernel_config->fused_sigmoid == 1) {
+      const char i_vname = (io_generated_code->arch < LIBXSMM_X86_AVX512) ? 'y' : 'z';
```
Comment (alheinecke) [inline src/generator_gemm_amx_microkernel_emu.c:189]: This condition cannot be true

**Review by alheinecke — CHANGES_REQUESTED** [review]
What about the reduce and vecopreduce kernels have you verified they go got AVX2?

File: src/generator_gemm_amx_microkernel_emu.c (hunk 3 above)
Comment (deeptiag1) [reply src/generator_gemm_amx_microkernel_emu.c]: Resolved it

File: src/generator_gemm_amx_microkernel.c (hunk 2 above)
Comment (deeptiag1) [reply src/generator_gemm_amx_microkernel.c]: fixed

File: src/generator_gemm_amx_microkernel.c (hunk 1 above)
Comment (deeptiag1) [reply src/generator_gemm_amx_microkernel.c]: fixed

**Pull request merged**